source/javascripts/application.js

```diff
@@ -1,5 +1,40 @@
+function submitCookiePreferences(e) {
+  e.preventDefault();
+
+  const data = new FormData(this);
+
+  for (let category of data.keys()) {
+    GovWifi.cookies.setCategoryAllowed(category, data.get(category) === "on");
+  }
+
+  document.getElementById("cookie-preferences-amended").style.display = "block";
+
+  window.scrollTo(0, 0);
+}
+
+function setCategoryInput(category) {
```

Comment:
Non-blocking suggestion - the name `setCategoryInput` seems a little obscure for what it does and from where it's being invoked, something like `initCategorySelection` or `setupCategorySelection` or `initActiveCategory` or `setupActiveCategory` would sound clearer to me. But this is subjective, and feel free to disagree.